weasyl/test/login/test_authenticate_bcrypt.py

```diff
@@ -0,0 +1,140 @@
+# This Python file uses the following encoding: utf-8
+"""
+Test suite for: login.py::def authenticate_bcrypt(username, password, session=True):
+"""
+import json
+import web
+
+from weasyl.test import db_utils
+from weasyl import login, macro, media
+from weasyl import define as d
+
+
+# Main test password
+raw_password = "0123456789"
+
+
+class Bag(object):
+    def __init__(self, **kw):
+        for kv in kw.items():
+            setattr(self, *kv)
+
+
+def configure_moderators(mod_user_id):
+    """ Used to set staff.py's global MODS variable """
+    from libweasyl.configuration import configure_libweasyl
+
+    configure_libweasyl(
+        dbsession=d.sessionmaker,
+        not_found_exception=web.notfound,
+        base_file_path='testing',
+        staff_config_dict={"mods": [mod_user_id]},
+        media_link_formatter_callback=media.format_media_link,
+    )
+
+
+def test_logins_fail_if_no_username_provided():
+    result = login.authenticate_bcrypt(username='', password='password')
+    assert result == (0, 'invalid')
+
+
+def test_logins_fail_if_no_password_provided():
+    user_name = "NoPasswordProvided"
```

Comment:
Use the same username (e.g. `"test"`) for all of these unless the test requires something else, please.